ios/AirInAppPurchase/AirInAppPurchase.m

```diff
@@ -114,18 +116,31 @@ - (void)productsRequest:(SKProductsRequest *)request didReceiveResponse:(SKProdu
     
     [dictionary setObject:productElement forKey:@"details"];
     
-    
-    NSString* jsonDictionary = [dictionary JSONString];
-    
-    FREDispatchStatusEventAsync(AirInAppCtx ,(uint8_t*) "PRODUCT_INFO_RECEIVED", (uint8_t*) [jsonDictionary UTF8String] );
-    
-    if ([response invalidProductIdentifiers] != nil && [[response invalidProductIdentifiers] count] > 0)
+    if ([NSJSONSerialization isValidJSONObject:dictionary])
     {
-        NSString* jsonArray = [[response invalidProductIdentifiers] JSONString];
+        NSData *json;
+        NSError *error = nil;
         
-        FREDispatchStatusEventAsync(AirInAppCtx ,(uint8_t*) "PRODUCT_INFO_ERROR", (uint8_t*) [jsonArray UTF8String] );
+        // Serialize the dictionary
+        json = [NSJSONSerialization dataWithJSONObject:dictionary options:NSJSONWritingPrettyPrinted error:&error];
         
```

Comment:
I don't think the block for PRODUCT_INFO_ERROR should be dependent on the success of the json serialization. If anything it should fire off if the json fails.